homeassistant/components/gios/__init__.py

```diff
@@ -0,0 +1,91 @@
+"""The GIOS component."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from aiohttp.client_exceptions import ClientConnectorError
+from async_timeout import timeout
+from gios import ApiError, Gios, NoStationError
+
+from homeassistant.core import Config, HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.util import Throttle
+
+from .const import CONF_STATION_ID, DATA_CLIENT, DOMAIN
+
+# Term of service GIOŚ allow downloading data no more than twice an hour.
+DEFAULT_SCAN_INTERVAL = timedelta(minutes=30)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: Config) -> bool:
+    """Set up configured GIOS."""
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][DATA_CLIENT] = {}
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up GIOS as config entry."""
+    station_id = config_entry.data[CONF_STATION_ID]
+    _LOGGER.debug("Using station_id: %s", station_id)
+
+    websession = async_get_clientsession(hass)
+
+    gios = GiosData(websession, station_id)
+
+    await gios.async_update()
+
+    hass.data[DOMAIN][DATA_CLIENT][config_entry.entry_id] = gios
+
+    config_entry.add_update_listener(update_listener)
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(config_entry, "air_quality")
+    )
+    return True
+
+
+async def async_unload_entry(hass, config_entry):
+    """Unload a config entry."""
+    hass.data[DOMAIN][DATA_CLIENT].pop(config_entry.entry_id)
+    await hass.config_entries.async_forward_entry_unload(config_entry, "air_quality")
+    return True
+
+
+async def update_listener(hass, entry):
+    """Update listener."""
+    await hass.config_entries.async_forward_entry_unload(entry, "air_quality")
+    hass.async_add_job(
+        hass.config_entries.async_forward_entry_setup(entry, "air_quality")
+    )
```

Comment:
Why would you do this?

Reply:
This is a left over of the config_flow options that I removed. I think I'm blind. I will remove it.